src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs

```diff
@@ -524,7 +536,8 @@ private bool EvaluateHttpSourceUsage()
             return !error;
         }
 
-        private async Task<(bool, bool, string, PackagesLockFile)> EvaluateLockFile(TelemetryActivity telemetry, RemoteWalkContext contextForProject, string packagesLockFilePath, PackagesLockFile packagesLockFile, CancellationToken token)
+        private async Task<(bool, bool, bool, string, PackagesLockFile)>
```

Comment:
It might be time to rethink the return type of this method. A private class or struct could help add structure to the results rather than a tuple. This is a suggestion so if you decide to defer for later, I will support that decision, especially because this is a private method.